src/zcl_att_span.clas.abap

```diff
@@ -1,24 +1,44 @@
 CLASS zcl_att_span DEFINITION
   PUBLIC
   FINAL
-  CREATE PUBLIC .
+  CREATE PRIVATE
+  GLOBAL FRIENDS zcl_att_trace_transaction.
 
   PUBLIC SECTION.
 
+    METHODS constructor
+      IMPORTING parent_span_id TYPE sysuuid_x16 OPTIONAL.
+
     METHODS add_span
-      RETURNING VALUE(span) TYPE REF TO zcl_att_span.
+      IMPORTING
+                parent_span_id TYPE sysuuid_x16
+      RETURNING VALUE(span)    TYPE REF TO zcl_att_span.
 
   PROTECTED SECTION.
   PRIVATE SECTION.
 
+    DATA span_id TYPE sysuuid_x16.
+
 ENDCLASS.
 
 
 
 CLASS zcl_att_span IMPLEMENTATION.
 
+  METHOD constructor.
+
+    TRY.
+        me->span_id = cl_system_uuid=>create_uuid_x16_static( ).
```

Comment:
@JohannesKonings do you know why me-> is a 'bad' thing?

Reply:
Because the abaplint rule for it still exists :)
We should remove the rule because we are not using prefixes and that's why we need me->.
We should adapt the overall rule setting soon and tidy up the abaplint issues.